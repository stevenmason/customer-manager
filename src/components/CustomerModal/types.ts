import { Customer } from "../../store/types";
import {
  setEditingCustomer,
  setEditingCustomerValue,
  setCustomer
} from "../../store/actions";

interface Props {
  customer: Customer;
  setEditingCustomer: typeof setEditingCustomer;
  setEditingCustomerValue: typeof setEditingCustomerValue;
  setCustomer: typeof setCustomer;
}

export default Props;
