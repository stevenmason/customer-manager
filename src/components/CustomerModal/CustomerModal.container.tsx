import { connect } from "react-redux";
import { bindActionCreators, Dispatch, Action } from "redux";

import {
  setEditingCustomer,
  setEditingCustomerValue,
  setCustomer
} from "../../store/actions";
import { editingCustomer } from "../../store/selectors";
import { State } from "../../store/types";
import CustomerModal from "./CustomerModal";

const mapStateToProps = (state: State) => ({
  customer: editingCustomer(state)!
});

const mapDispatchToProps = (dispatch: Dispatch<Action>) => ({
  ...bindActionCreators(
    {
      setEditingCustomer,
      setEditingCustomerValue,
      setCustomer
    },
    dispatch
  )
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(CustomerModal);
