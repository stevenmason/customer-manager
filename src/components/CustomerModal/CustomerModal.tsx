import React from "react";

import Props from "./types";
import Modal, {
  ModalBody,
  ModalTitle,
  ModalActions
} from "../../patterns/Modal";
import Input from "../../patterns/Input";
import Button from "../../patterns/Button";

const CustomerModal: React.FC<Props> = ({
  customer,
  setEditingCustomer,
  setEditingCustomerValue,
  setCustomer
}) => {
  const { firstName, lastName, dateOfBirth } = customer;
  return (
    <Modal onClose={() => setEditingCustomer(null)}>
      <ModalTitle>Customer:</ModalTitle>
      <ModalBody>
        <Input
          name="firstName"
          value={firstName}
          label="First Name"
          onChange={setEditingCustomerValue}
        />
        <Input
          name="lastName"
          value={lastName}
          label="Last Name"
          onChange={setEditingCustomerValue}
        />
        <Input
          type="date"
          name="dateOfBirth"
          value={dateOfBirth}
          label="Date of Birth"
          onChange={setEditingCustomerValue}
        />
      </ModalBody>
      <ModalActions>
        <Button onClick={() => setEditingCustomer(null)}>Cancel</Button>
        <Button
          onClick={() => {
            setCustomer(customer);
            setEditingCustomer(null);
          }}
        >
          Save
        </Button>
      </ModalActions>
    </Modal>
  );
};

export default CustomerModal;
