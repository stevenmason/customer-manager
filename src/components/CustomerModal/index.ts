import CustomerModal from "./CustomerModal.container";
export default CustomerModal;
