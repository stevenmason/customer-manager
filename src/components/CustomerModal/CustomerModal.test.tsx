import React from "react";
import renderer from "react-test-renderer";

import CustomerModal from "./CustomerModal";
import {
  setEditingCustomer,
  setCustomer,
  setEditingCustomerValue
} from "../../store/actions";

const PROPS = {
  customer: {
    id: "1",
    firstName: "Steven",
    lastName: "Mason",
    dateOfBirth: ""
  },
  setEditingCustomer,
  setCustomer,
  setEditingCustomerValue
};

it("renders correctly", () => {
  const tree = renderer.create(<CustomerModal {...PROPS} />).toJSON();
  expect(tree).toMatchSnapshot();
});
