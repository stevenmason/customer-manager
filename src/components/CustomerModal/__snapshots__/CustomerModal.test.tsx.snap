// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-bdVaJa eksIDc"
>
  <div
    className="sc-htpNat jabHwM"
    onClick={[Function]}
  />
  <div
    className="sc-bwzfXH hXDBTd"
  >
    <div
      className="sc-ifAKCX hcvhgT"
    >
      <h3>
        Customer:
      </h3>
    </div>
    <div
      className="sc-bxivhb eIcZDB"
    >
      <div
        className="sc-bZQynM cicyuq"
        size="LARGE"
      >
        <label
          className="sc-gzVnrw gBYcjp"
        >
          First Name
        </label>
        <input
          className="sc-htoDjs iRTnix"
          name="firstName"
          onChange={[Function]}
          value="Steven"
        />
      </div>
      <div
        className="sc-bZQynM cicyuq"
        size="LARGE"
      >
        <label
          className="sc-gzVnrw gBYcjp"
        >
          Last Name
        </label>
        <input
          className="sc-htoDjs iRTnix"
          name="lastName"
          onChange={[Function]}
          value="Mason"
        />
      </div>
      <div
        className="sc-bZQynM cicyuq"
        size="LARGE"
      >
        <label
          className="sc-gzVnrw gBYcjp"
        >
          Date of Birth
        </label>
        <input
          className="sc-htoDjs iRTnix"
          name="dateOfBirth"
          onChange={[Function]}
          type="date"
          value=""
        />
      </div>
    </div>
    <div
      className="sc-EHOje dSjNu"
    >
      <button
        className="sc-dnqmqq keCVKD"
        onClick={[Function]}
      >
        Cancel
      </button>
      <button
        className="sc-dnqmqq keCVKD"
        onClick={[Function]}
      >
        Save
      </button>
    </div>
  </div>
</div>
`;
